### DISCUSSION TITLE
CMT2300A Frequenz Konfiguration Einstellungen

### DISCUSSION THREAD

**Discussion by Manos1966** [hoylabs/OpenDTU-OnBattery#1219] [Q&A]
Anlass ist eine Diskussion die ich gefunden habe: 
> Ich konnte das Problem mit der Verbindung mit dem HMT lösen indem ich die Frequenz geändert habe auf 686.25MHz.

Ich habe nirgendwo beschrieben gefunden, welche CMT2300A Frequenz Einstellungen wir brauchen in der OpenDTU-OnBattery!

Die DTU-Einstellungen haben Optionen fuer Sendeleistung, Region/Land, Frequenz.
Die ersten zwei sind ausführlich diskutiert, wie sieht es aus aber mit Frequenz?

![CMT2300A DTU Einstellungen](https://github.com/user-attachments/assets/637e253f-c381-45e1-9cb4-70340303ba0e)

Laut einer Diskussion hier:  https://github.com/tbnobody/OpenDTU/discussions/1300

> Die Google AI sagt mir: "Die Hoymiles HMS-Serie verwendet das 868-MHz-Band für die Kommunikation zwischen den Wechselrichtern und dem Gateway. Die Geräte sind auf die Verwendung des Subbands von 868,0 MHz bis 868,6 MHz mit einer Bandbreite von 600 kHz optimiert. Die maximale Sendeleistung der Geräte beträgt 10 mW EIRP.

> Initial wird IMMER auf 868MHZ kommuniziert. Mit dem ersten Paket muss die DTU die Frequenz des Inverters einstellen. Das ist die Frequenz die in den Einstellungen dargestellt ist (865). Muss den Bandplänen usw. entsprechen.

> Falls dieses Problem noch bei einigen aktuell ist, ich hatte bis gestern auch massive Probleme.
Kurz zu meiner Geschichte:
Ich habe seit Oktober ´23 zwei Wechselrichter (HMS-1600-4T)am laufen und habe mir dazu die AHOY-DTU gebaut (großes Lob an die Erfinder und Entwickler). Sporadisch hatte ich an einigen Tagen sehr schlechten Empfang zwischen Wechselrichter und AHOY. Die Fehlerrate (keine Antwort) war jenseits von 60%. Ich habe alles versucht, vom Neustart der DTU und Wechselrichter, über Up- und Downgrade der AHOY-Version, bis zum Trennen des Stroms, aber an diesem Tag konnte ich die Qualität der Verbindung nicht verbessern. Am nächsten Tag war meistens wieder alles OK (+-15% Verlustrate). Dies hielt dann wieder ein paar Tage an, um dann wieder einen ganzen Tag schlecht zu laufen. Natürlich habe ich versucht, die Frequenz zu ändern, aber die DTU baute keine Verbindung zum WR auf, worauf ich dann wieder die alte Frequenz eingestellt hatte, damit ich wenigstens etwas Empfang wiederherstellen konnte.
Die Lösung kam gestern, als ich mal wieder sehr schlechten Empfang hatte:
Ich habe in der DTU nochmals die Frequenzen geändert. Jeder WR hat jetzt seine eigene Frequenz. Vorher waren alle bei 863 MHz und jetzt bei 866 und 868 MHz. Man muss allerdings nach Änderung der Frequenz 10 Min warten und dabei die AHOY-DTU einfach laufen lassen (kann auch 15 Min gewesen sein). Die Wechselrichter übernehmen dann die Frequenz und senden auf dieser weiter.
Ich hatte schlagartig 0% Verlustrate und dementsprechend besten Empfang. Heute habe ich mal geguckt und ein WR ist bei 0,41% Verlust und der andere bei knapp 4% was aber noch völlig im grünen Bereich ist.
Ich kann nur vermuten, wieso es bei 863 MHz so schlecht lief. Vielleicht hat irgendein Nachbar auch ne DTU oder ne Alarmanlage oder nen Rasenmäher oder was weiß ich, welcher auch auf derselben Frequenz funkt und hat damit die Störungen in meiner DTU verursacht. Jedenfalls läuft es derzeit sehr gut.
Ich hoffe, meine Lösung kann noch weiteren Leuten bei Verbindungsproblemen helfen. Vielleicht sollte das Thema "Frequenzumstellung" mal in der AHOY Anleitung unter "Troubleshooting" mit aufgenommen werden. Bei Google findet man nämlich garnichts. Auch nicht, dass man nach Umstellung 10-15 Min warten muss, sonst hätte ich das schon viel eher versucht.

**Comment by edddymax** [comment]
Hatte in den letzten 2-3Tagen massive Verbindungsprobleme ( ESP32-s3 —- HMS 1600 4T).Habe die aktuelle Firmware installiert und bin von ESP32 zu …s3 umgestiegen. Die Änderung der Frequenz von 865,00 zu 868,00 hat zum Erfolg geführt. Läuft jetzt erst seit 5 Stunden. Mal sehen, wie stabil die Verbindung bestehen bleibt. Kann gerne meine pin_Mapping mitteilen, wenn es jemand braucht.

**Reply by hoschiking** [reply]
Wir reden hier von der HM Serie, die funken auf 2,4 GHz. Mit nur einem Wechselrichter läuft es seit einem Monat ohne Probleme, mit 2 vielleicht nur einen Tag stabil. Eine schlechte Verbindung wäre noch das kleinste Problem, sobald die Verbindung abbricht geht gar nichts mehr, nur ein reboot feet dtu bringt Abhilfe.
Es sieht für mich nach einem Bug in der dtu aus, wenn auch ich es nicht belegen kann. Jedoch nach diversen Tests ist es die wahrscheinlichste Variante

**Reply by Manos1966** [reply]
Nein @hoschiking  wir reden von der HMS- und HMT- Serien. Sie haben nichts mit NRF24+ (2,4GHz) zu tun.

**Reply by hoschiking** [reply]
Ups, falscher Thread. Sorry dafür. 
Der Eddymax hat in einem von mir erstellten Thread auf diesen verlinkt, aber in meinem geht um eben um die HM Serie, und da passt der Lösungsansatz nicht.
